Runtime/Components/CinemachineSplineDolly.cs

```diff
@@ -0,0 +1,494 @@
+            /// </summary>
+            [HideInInspector]
+            [Tooltip("Affects how many segments to split a spline into when calculating the nearest point." +
+                "Higher values mean smaller and more segments, which increases accuracy at the cost of processing time. " +
+                "In most cases, the default value (4) is appropriate. Use with SearchIteration to fine-tune point accuracy.")]
+            public int m_SearchResolution;
+
+            /// <summary>
+            /// The nearest point is calculated by finding the nearest point on the entire length
+            /// of the spline using <seealso cref="m_SearchResolution"/> to divide into equally spaced line segments. Successive
+            /// iterations will then subdivide further the nearest segment, producing more accurate results. In most cases,
+            /// the default value is sufficient.
+            /// For more information, see SplineUtility.GetNearestPoint.
+            /// </summary>
+            [HideInInspector]
+            [Tooltip("The nearest point is calculated by finding the nearest point on the entire length of the spline " +
+                "using SearchResolution to divide into equally spaced line segments. Successive iterations will then " +
+                "subdivide further the nearest segment, producing more accurate results. In most cases, the default value (2) is sufficient.")]
+            public int m_SearchIteration;
+        }
+
+        /// <summary>Controls how automatic dollying occurs</summary>
+        [Tooltip("Controls how automatic dollying occurs.  A Follow target is necessary to use this feature.")]
+        public AutoDolly m_AutoDolly = new AutoDolly 
+        { 
+            m_Enabled = false, 
+            m_PositionOffset = 0,
+            m_SearchIteration = 4, 
+            m_SearchResolution = 2,
+        };
+
+        /// <summary>True if component is enabled and has a spline</summary>
+        public override bool IsValid => enabled && m_Spline != null;
+
+        /// <summary>Get the Cinemachine Pipeline stage that this component implements.
+        /// Always returns the Body stage</summary>
+        public override CinemachineCore.Stage Stage => CinemachineCore.Stage.Body;
+
+        /// <summary>
+        /// Report maximum damping time needed for this component.
+        /// </summary>
+        /// <returns>Highest damping setting in this component</returns>
+        public override float GetMaxDampTime() => !m_DampingEnabled ? 0 :
+            Mathf.Max(Mathf.Max(m_Damping.x, Mathf.Max(m_Damping.y, m_Damping.z)), m_AngularDamping);
+
+        /// <summary>
+        /// Adds spline to SplineDolly.
+        /// </summary>
+        /// <param name="spline">The spline to add. If null, then any spline on SplineDolly will be removed.</param>
+        public void AddSplineContainer(SplineContainer spline)
+        {
+            m_Spline = spline;
+            UpdateSplineData();
+        }
```

Comment:
I needed to add this to avoid recomputing spline length and knot count. This might not be needed if Splines will have a built in input sanitizer.